bootstrap/kubeadm/internal/cloudinit/cloudinit_test.go

```diff
@@ -94,6 +95,9 @@ func TestNewInitControlPlaneCommands(t *testing.T) {
 
 	cpinput := &ControlPlaneInput{
 		BaseUserData: BaseUserData{
+			BootCommands: []bootstrapv1.BootCommand{
+				{"echo", "hello"},
```

Comment:
In order to make this test to be useful to validate the code actually works/to prevent regressions, you should also modify test assertions down below that boot commands actually show up in the generated output (and possibly, ensuring that command show up in the right section of the generated file)

Reply:
I wasn't sure how to ensure that the commands show up in the right section of the generated file, but I added a check that the bootcommands show up in the generated file and also that the `bootcmd:` string is there (which is omitted when bootCommands are unset)

https://github.com/kubernetes-sigs/cluster-api/pull/11271/commits/a4f3bdbc53b27837ca95cd7ca67419b2e0afe85d

What do you think?

Reply:
Thanks for the suggestion!

https://github.com/kubernetes-sigs/cluster-api/pull/11271/commits/6fb468d53a18804bc2b585138a12a68f008efa7e